src/common/interval.h

```diff
@@ -18,6 +18,14 @@ class interval {
     interval(const interval& a, const interval& b)
       : min(fmin(a.min, b.min)), max(fmax(a.max, b.max)) {}
 
+    double size() const {
+        return max - min;
+    }
+
+    interval expand(double delta) const {
+        return interval(min - delta, max + delta);
```

Comment:
should this be:
```
        return interval(min - delta/2, max + delta/2);
```

Reply:
since it checks to see if it's less than delta, and if it is, makes it twice delta. Anything from 1 to 2 delta will remain as is

Reply:
Nice catch; will fix.